api/sdk/sdk.go

```diff
@@ -60,166 +56,208 @@ type memberKeys struct {
-	memberName := testutils.RandomString()
 	ks := platformpolicy.NewKeyProcessor()
 
 	privateKey, err := ks.GeneratePrivateKey()
 	if err != nil {
-		return nil, "", errors.Wrap(err, "[ CreateMember ] can't generate private key")
+		return nil, "", errors.Wrap(err, "failed to generate private key")
 	}
 
-	privateKeyStr, err := ks.ExportPrivateKeyPEM(privateKey)
+	privateKeyBytes, err := ks.ExportPrivateKeyPEM(privateKey)
 	if err != nil {
-		return nil, "", errors.Wrap(err, "[ CreateMember ] can't export private key")
+		return nil, "", errors.Wrap(err, "failed to export private key")
 	}
+	privateKeyStr := string(privateKeyBytes)
 
 	memberPubKeyStr, err := ks.ExportPublicKeyPEM(ks.ExtractPublicKey(privateKey))
 	if err != nil {
-		return nil, "", errors.Wrap(err, "[ CreateMember ] can't extract public key")
+		return nil, "", errors.Wrap(err, "failed to extract public key")
 	}
 
-	params := []interface{}{memberName, string(memberPubKeyStr)}
-	body, err := sdk.sendRequest(ctx, "CreateMember", params, sdk.rootMember)
+	response, err := sdk.DoRequest(
+		sdk.rootMember.Caller,
+		privateKeyStr,
+		"contract.createMember",
+		map[string]interface{}{"publicKey": string(memberPubKeyStr)},
+	)
 	if err != nil {
-		return nil, "", errors.Wrap(err, "[ CreateMember ] can't send request")
+		return nil, "", errors.Wrap(err, "request was failed ")
 	}
 
-	response, err := sdk.getResponse(body)
-	if err != nil {
-		return nil, "", errors.Wrap(err, "[ CreateMember ] can't get response")
-	}
+	return NewMember(response.ContractResult.(string), privateKeyStr), response.TraceID, nil
+}
 
-	if response.Error != "" {
-		return nil, response.TraceID, errors.New(response.Error)
+// AddBurnAddresses method add burn addresses
+func (sdk *SDK) AddBurnAddresses(burnAddresses []string) (string, error) {
+	response, err := sdk.DoRequest(
+		sdk.migrationAdminMember.Caller,
+		sdk.migrationAdminMember.PrivateKey,
+		"wallet.addBurnAddresses",
+		map[string]interface{}{"burnAddresses": burnAddresses},
+	)
+	if err != nil {
+		return "", errors.Wrap(err, "request was failed ")
 	}
 
-	return NewMember(response.Result.(string), string(privateKeyStr)), response.TraceID, nil
+	return response.TraceID, nil
 }
 
 // Transfer method send money from one member to another
-func (sdk *SDK) Transfer(amount uint, from *Member, to *Member) (string, error) {
-	ctx := inslogger.ContextWithTrace(context.Background(), "Transfer")
-	params := []interface{}{amount, to.Reference}
-	config, err := requester.CreateUserConfig(from.Reference, from.PrivateKey)
+func (sdk *SDK) Transfer(amount string, from *Member, to *Member) (string, error) {
+	response, err := sdk.DoRequest(
+		from.Reference,
+		from.PrivateKey,
+		"wallet.transfer",
+		map[string]interface{}{"amount": amount, "to": to.Reference},
+	)
 	if err != nil {
-		return "", errors.Wrap(err, "[ Transfer ] can't create user config")
+		return "", errors.Wrap(err, "request was failed ")
```

Comment:
please add context to errors, i.e. here:
'transfer request was failed'

Reply:
it will be excess data.